rust/src/io/reader.rs

```diff
@@ -178,14 +183,20 @@ impl FileReader {
             read_struct_from_buf(&tail_bytes.slice(offset..))?
         };
 
-        let (projection, num_columns) = if let Some(m) = manifest {
-            (m.schema.clone(), m.schema.max_field_id().unwrap() + 1)
+        let mut m: Manifest;
+        let manifest = if let Some(m) = manifest {
+            m
         } else {
-            let mut m: Manifest =
-                read_struct(object_reader.as_ref(), metadata.manifest_position.unwrap()).await?;
+            m = read_struct(object_reader.as_ref(), metadata.manifest_position.unwrap()).await?;
             m.schema.load_dictionary(object_reader.as_ref()).await?;
+            &m
+        };
 
-            (m.schema.clone(), m.schema.max_field_id().unwrap() + 1)
+        let (projection, num_columns) = {
+            (
+                manifest.schema.clone(),
+                manifest.schema.max_field_id().unwrap() + 1,
+            )
         };
```

Comment:
nit: I think this could be simple assignment now
```suggestion
        let projection = manifest.schema.clone();
        let num_columns = manifest.schema.max_field_id().unwrap() + 1;
```